File: O(n)/Linear.java
public class Linear {
    public static int buscaLinear(int[] array, int alvo) {
        // Percorre de 0 até n-1
        for (int i = 0; i < array.length; i++) { // (O(n) iterações
            if (array[i] == alvo) {
                return i; // Encontrou
            }
        }
        return -1; // Não encontrou
    }
        public static void main(String[] args) {
        int[] numeros = {10, 50, 30, 70, 80, 20};
        System.out.println("Índice: " + buscaLinear(numeros, 30));
    }
}

File: O(n)/linear.py
def busca_linear(lista, alvo):
    # O(n): O laço percorre cada elemento da lista
    for i in range(len(lista)): # O(n) iterações
        if lista[i] == alvo:
            return i  # Retorna o índice encontrado
    return -1  # Retorna -1 se não encontrar

# Teste
numeros = [10, 50, 30, 70, 80, 20]
alvo = 30
indice = busca_linear(numeros, alvo)
print(f"Elemento encontrado no índice: {indice}")